### PR TITLE
[kube-prometheus-stack] Update mixins, switch to kube-prometheus mixin and fix label aggregation in TargetDown alert

### PR DESCRIPTION
**Pull request by jkroepke**
#### What this PR does / why we need it

Switch to jsonnet which generates all rules now by default (including the new grafana one)

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-prometheus-stack/hack/sync_prometheus_rules.py [hunk 1]
```diff
@@ -105,6 +112,8 @@ def new_representer(dumper, data):
     'config-reloaders': ' .Values.defaultRules.rules.configReloaders',
     'etcd': ' .Values.kubeEtcd.enabled .Values.defaultRules.rules.etcd',
     'general.rules': ' .Values.defaultRules.rules.general',
+    'grafana_rules': ' .Values.defaultRules.rules.grafana',
+    'GrafanaAlerts': ' .Values.defaultRules.rules.grafana',
```
Comment (GMartinez-Sisti) [inline charts/kube-prometheus-stack/hack/sync_prometheus_rules.py:116]: Two questions:
- are these supposed to be the same?
- can we rename `GrafanaAlerts` to `grafana-alerts` to make it similar with the other names and filenames?

Comment (jkroepke) [reply charts/kube-prometheus-stack/hack/sync_prometheus_rules.py:116]: > can we rename GrafanaAlerts to grafana-alerts to make it similar with the other names and filenames?

I saw this, too. Currently, the file name based on the rule group name, which behave differently at Grafana. But i can add a replacement for this, like to snake case.

> are these supposed to be the same?

GrafanaAlerts has an dependency against grafana_rules. And there are just 2 rules in total. I thought, its fine to group them. But I'm also open separate them.

No strong opinion here.

File: charts/kube-prometheus-stack/hack/sync_prometheus_rules.py [hunk 2]
```diff
@@ -29,49 +29,47 @@ def new_representer(dumper, data):
...
 
 # Source files list
 charts = [
     {
-        'source': 'https://raw.githubusercontent.com/prometheus-operator/kube-prometheus/%s/manifests/alertmanager-prometheusRule.yaml' % (refs['ref.kube-prometheus'],),
+        'git': 'https://github.com/prometheus-operator/kube-prometheus.git',
+        'branch': refs['ref.kube-prometheus'],
+        'source': 'main.libsonnet',
+        'cwd': 'jsonnet/kube-prometheus',
         'destination': '../templates/prometheus/rules-1.14',
```
Comment (mrueg) [inline charts/kube-prometheus-stack/hack/sync_prometheus_rules.py:46]: should we remove the -1.14 postfix?

Comment (jkroepke) [reply charts/kube-prometheus-stack/hack/sync_prometheus_rules.py:46]: It feel like it make sense, maybe it's something for an follow-up PR?

**Comment by GMartinez-Sisti** [comment]
Hi @jkroepke, sorry it took too long to review :/ please fix the conflicts and I'll take another look.

**Comment by jkroepke** [comment]
@QuentinBisson @GMartinez-Sisti i would much appreciate a review here.

**Comment by GMartinez-Sisti** [comment]
> @QuentinBisson @GMartinez-Sisti i would much appreciate a review here.

Hi! Looks like you need to generate the files again, maybe something got merged in between that changed them?

**Comment by jkroepke** [comment]
Done

**Review by GMartinez-Sisti — APPROVED** [review]

**Pull request merged**